Source/ARTRealtime.m

```diff
@@ -283,7 +286,7 @@ - (void)auth:(ARTAuthInternal *)auth didAuthorize:(ARTTokenDetails *)tokenDetail
         default: {
             // Client state is NOT Connecting or Connected, so it should start a new connection
             [self.logger debug:__FILE__ line:__LINE__ message:@"RS:%p new connection from successfull authorize %@", self.rest, tokenDetails];
-            [self transition:ARTRealtimeConnecting];
+            [self transition:ARTRealtimeConnecting withMetadata:[[ARTConnectionStateChangeMetadata alloc] init]];
```

Comment:
Why not to call `transition:` overload here and further in case of an empty `ARTConnectionStateChangeMetadata`?

Reply:
As I think I mentioned in one of the commit messages, I wanted to get rid of the overloads and have a single method.

I did consider allowing `nil` metadata, but decided it simpler in the end to just always require a metadata (especially since in the future it might be useful to give the metadata an ID that can be used for logging).

Reply:
I would consider to introduce a new static `empty` method then.

Reply:
I'm curious why? Given it's a type all of whose properties are nullable, it seems pretty intuitive to me how a bare `init` method would behave.

Reply:
It's more about style, I don't insist on that of course, `[ARTConnectionStateChangeMetadata empty]` looks more prettier then `[[ARTConnectionStateChangeMetadata alloc] init]]` imho.

Reply:
I think I'm going to keep it the way it is, if you don't mind?

Reply:
Sure, it's cosmetics.